config/user-pam.m4

```diff
@@ -12,8 +13,9 @@ AC_DEFUN([ZFS_AC_CONFIG_USER_PAM], [
 
 	AC_ARG_WITH(pamconfigsdir,
 		AS_HELP_STRING([--with-pamconfigsdir=DIR],
-		[install pam-config files in dir [[/usr/share/pamconfigs]]]),
-		[pamconfigsdir="$withval"],[pamconfigsdir=/usr/share/pam-configs])
+		[install pam-config files in dir [[$acl_final_prefix/share/pam-configs]]]),
+		[pamconfigsdir="$withval"],
+		[pamconfigsdir=$acl_final_prefix/share/pam-configs])
```

Comment:
I think `"${datarootdir}/pam-configs"` would be better?

The help messages also seem to have too many quotes. The output in `configure --help` looks like
`[[$libdir/security]]` or `[[/usr/share/pam-configs]]` while the standard ones (--libdir etc) would use `[LIBDIR/security]` etc.

Unrelated to this but the zfsexecdir one looks particularly weird: `[[[libexecdir]/zfs]]`

Reply:
or maybe `"${datadir}/pam-configs"`.

https://www.gnu.org/prep/standards/html_node/Directory-Variables.html

Reply:
Okay I was starting to get there!  saw a bunch of stuff from tests gets installed to share using foo_DATA ;)

You're right about the weird quoting, that could stand to be cleaned up.

Thanks!

Reply:
Sorry, with single quotes, not double. Otherwise it gets expanded one level at configure time, zfsexecdir has that problem: it's currently `${exec_prefix}/libexec/zfs` instead of `${libexecdir}/zfs` in the Makefile.

Reply:
I'm sure I just did something silly with the message 😄